### PR TITLE
fix: add SHA1 key to the APK Build

### PR DESCRIPTION
**Pull request by LawrenceLeitgib**
this modifcation make the APK-builder sign the APK with the keystore that have been added to github secrect

### CODE REVIEW

File: .github/workflows/build-apk.yml [hunk 1]
```diff
@@ -48,13 +49,43 @@ jobs:
...
+      - name: Sign Debug APK with apksigner
+        run: |
+          ${ANDROID_HOME}/build-tools/${{ env.build_tools_version }}/apksigner sign --ks ~/.android/debug.keystore --ks-key-alias ${{ secrets.KEY_ALIAS }} --ks-pass pass:${{ secrets.KEYSTORE_PASSWORD }} --key-pass pass:${{ secrets.KEYSTORE_PASSWORD }} app/build/outputs/apk/debug/app-debug.apk
+
+      # Verify APK signature
+      - name: Verify APK Signature
+        run: |
+          ${ANDROID_HOME}/build-tools/${{ env.build_tools_version }}/apksigner verify app/build/outputs/apk/debug/app-debug.apk
+
+      # Verify APK Signature using SHA1
```
Comment (santiagoarayaramos) [inline .github/workflows/build-apk.yml:82]: This doesn't verify APK signature per se, instead it prints them, so allows for visual verification of SHA1. Specify in comment and step name to make it clearer maybe

File: .github/workflows/build-apk.yml [hunk 2]
```diff
@@ -48,13 +49,43 @@ jobs:
...
+      # Build the debug APK
       - name: Build APK
         run: ./gradlew assembleDebug
 
+      # Find the latest version of build-tools
+      - name: Find latest build-tools version
+        id: build_tools
+        run: echo "build_tools_version=$(ls ${ANDROID_HOME}/build-tools | sort -V | tail -n 1)" >> $GITHUB_ENV
+
+      # Sign the debug APK using apksigner
```
Comment (santiagoarayaramos) [inline .github/workflows/build-apk.yml:72]: An eventual [zipalign](https://developer.android.com/tools/zipalign) could be added to improve memory efficiency, to be done before signing as specified on [apksigner](https://developer.android.com/tools/apksigner)

File: .github/workflows/build-apk.yml [hunk 3]
```diff
@@ -48,13 +49,43 @@ jobs:
...
+      # Decode the Keystore from secret
+      - name: Decode Keystore
+        env:
+          ANDROID_KEYSTORE_BASE64: ${{ secrets.ANDROID_KEYSTORE_BASE64 }}
+        run: |
+          mkdir -p ~/.android
+          echo "$ANDROID_KEYSTORE_BASE64" | base64 --decode > ~/.android/debug.keystore
+
+        # Generate SHA1 fingerprint from keystore
+      - name: Generate SHA1
```
Comment (santiagoarayaramos) [inline .github/workflows/build-apk.yml:61]: I believe this steps shows the content of the keystone, it doesn't generate the SHA1. This is already stored as a key in the key store, and applied when calling apksigner. This is useful for debugging, and making sure everything is in order, but for a proper release not necessary. 
If left, rename the step (if my interpretation is correct)

**Review by santiagoarayaramos — APPROVED** [review]
Good code, good work! Added a few comments to modify some names and remember to remove steps useful for debugging while we are still developing the app that could be removed on release, and some potential future improvements. All functionalities are working, approved!

File: .github/workflows/build-apk.yml (hunk 3 above)
Comment (LawrenceLeitgib) [reply .github/workflows/build-apk.yml:61]: yes your interpretation is correct I forgot to change it, I will leave as it is but modify the name

File: .github/workflows/build-apk.yml (hunk 1 above)
Comment (LawrenceLeitgib) [reply .github/workflows/build-apk.yml]: yes, I will change that too

**Pull request merged**